pkg/manifests/manifests.go

```diff
@@ -3249,6 +3249,16 @@ func (f *Factory) ThanosQuerierDeployment(grpcTLS *v1.Secret, enableUserWorkload
 				d.Spec.Template.Spec.Containers[i].Args = append(d.Spec.Template.Spec.Containers[i].Args, fmt.Sprintf("--log.level=%s", f.config.ClusterMonitoringConfiguration.ThanosQuerierConfig.LogLevel))
 			}
 
+			if f.config.ClusterMonitoringConfiguration.ThanosQuerierConfig.LoggingConfig != nil {
+				d.Spec.Template.Spec.Containers[i].Args = append(d.Spec.Template.Spec.Containers[i].Args, "--request.logging-config=$(REQUEST_LOGGING_CONFIG)")
+				d.Spec.Template.Spec.Containers[i].Env = append(d.Spec.Template.Spec.Containers[i].Env, v1.EnvVar{
+					Name: "REQUEST_LOGGING_CONFIG",
+					ValueFrom: &v1.EnvVarSource{
+						ConfigMapKeyRef: f.config.ClusterMonitoringConfiguration.ThanosQuerierConfig.LoggingConfig,
```

Comment:
Will the config get updated if this confimap gets changed?

Reply:
I am not sure, but this draft is purely for testing purposes right now.

Reply:
@fpetkovski - to answer your question, changing the cm will not force a rollout of the querier pods.

Reply:
Ack, thanks for checking 👍